src/App.js

```diff
@@ -30,23 +31,111 @@ const App = () => {
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
 
+  const [currentSquare, setCurrentSquare] = useState(false);
+
   // Wave 2
+  const changeTheSquare = (id) => {
+    let newSquares = [];
+    for (let row = 0; row < 3; row += 1) {
+      newSquares.push([]);
+      for (let col = 0; col < 3; col += 1) {
+        if (id === squares[row][col].id) {
+          if (!squares[row][col].value) {
+            if (!currentSquare) {
+              squares[row][col].value = PLAYER_1;
+            } else {
+              squares[row][col].value = PLAYER_2;
+            }
+            setCurrentSquare(!currentSquare);
+          }
+        }
+        newSquares[row].push(squares[row][col]);
+      }
+    }
+    setSquares(newSquares);
+  };
+
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
   const checkForWinner = () => {
```

Comment:
This looks like it works to detect a winner.  The only problem is that you're using `'X'` instead of `'x'` etc